docs/develop/tools-and-features/gsoc.md

```diff
@@ -0,0 +1,312 @@
+
+The service node determines the data used to derive the GSOC private key. Any node with access to this data can derive the same key and update the GSOC. The service node also defines a message format, filtering out non-conforming updates. Nodes which know both the GSOC key derivation data and the expected message format can send messages to the service node. Since only full nodes sync neighborhood chunks, the service node *must be a full node to receive GSOC updates*.  
+
+To receive messages in real time, the service node establishes a WebSocket connection to listen for GSOC update events. When a matching SOC update reaches a node with an active GSOC connection, an event is emitted, enabling the service node to dynamically receive messages as part of a many-to-one notification system.  
+
+:::info  
+GSOC was initially introduced in a [SWIP](https://github.com/ethersphere/SWIPs/blob/99e6cf90a4768b24d27e5339b205c18825b53322/SWIPs/swip-draft_graffiti-soc.md#gsoc-identifier), which outlines its core concepts and implementation details, and it is an evolution of the earlier [Graffiti feed](https://github.com/fairDataSociety/FIPs/blob/master/text/0062-graffiti-feed.md) feature.  
+:::  
+
+:::warning  
+The GSOC feature was introduced in Bee v2.3.0, along with the experimental [GSOC JS library](https://www.npmjs.com/package/@anythread/gsoc). Both are still in development and not recommended for production use.  
+:::  
+
+## GSOC Usage Guide
+
+While GSOC can be used by interacting directly with the `/gsoc/subscribe/{address}` endpoint, most users should use the [experimental GSOC library](https://www.npmjs.com/package/@anythread/gsoc), which simplifies the process (GSOC support will soon be added to [bee-js](/docs/develop/tools-and-features/bee-js/) as well).  
+
+The GSOC library provides the `InformationSignal` class, which establishes a WebSocket connection with a GSOC  and enables reading and writing to the GSOC over that connection.  
+
+- A **service node** uses `InformationSignal` to subscribe to GSOC update events.  
+- A **writer node** uses `InformationSignal` to update the GSOC in order to send messages that the service node can receive.  
+
+### Requirements
+
+To use GSOC, you need:  
+
+1. A full Bee node and a second light Bee node  
+2. A small amount of xBZZ and xDAI for postage stamp purchases and uploads  
+3. NodeJS & NPM  
+4. A mutable stamp batch (*set the* [`immutable` header parameter](/api/#tag/Postage-Stamps/paths/~1stamps~1%7Bamount%7D~1%7Bdepth%7D/post) *to `false` when* [buying a batch](/docs/develop/access-the-swarm/buy-a-stamp-batch#buying-a-stamp-batch))  
```

Comment:
Recommended, not a hard requirement. For full clarify maybe explain in a footnote why it is beneficial?

Reply:
Sorry it's right there :)